packages/docusaurus-plugin-redoc/src/index.ts

```diff
@@ -45,12 +45,13 @@ export default function redocPlugin(
   const { debug, spec, url: downloadUrl, config, themeId, normalizeUrl: normalizeDownloadUrl } = options;
 
   let url = downloadUrl;
-  const isSpecFile = fs.existsSync(spec);
+  const isExternalUrl = !!url;
+
   const fileName = path.join(
     'redocusaurus',
     `${options.id || 'api-spec'}.yaml`,
   );
-  let filesToWatch: string[] = isSpecFile ? [path.resolve(spec)] : [];
+  let filesToWatch: string[] = !isExternalUrl ? [path.resolve(spec)] : [];
```

Comment:
undoing the change in plugin, I think passing the spec directly can work without involving the plugin.

basically by passing no specs array to preset so only configure the theme and directly use it. no need to create plugin